components/tikv_util/src/config.rs

```diff
@@ -142,35 +142,32 @@ impl FromStr for ReadableSize {
             return Err(format!("ASCII string is expected, but got {:?}", s));
         }
 
-        let mut chrs = size_str.chars();
-        let mut number_str = size_str;
-        let mut unit_char = chrs.next_back().unwrap();
-        if unit_char < '0' || unit_char > '9' {
-            number_str = chrs.as_str();
-            if unit_char == 'B' {
-                let b = match chrs.next_back() {
-                    Some(b) => b,
-                    None => return Err(format!("numeric value is expected: {:?}", s)),
-                };
-                if b < '0' || b > '9' {
-                    number_str = chrs.as_str();
-                    unit_char = b;
-                }
+        // size: digits and '.' as decimal separator
+        let size = size_str
+            .to_string()
+            .chars()
+            .take_while(|c| char::is_ascii_digit(c) || *c == '.')
+            .collect::<String>();
```

Comment:
Seems you can use `count` to avoid the allocation here.

Reply:
Good catch